pandas/tests/groupby/test_groupby.py

```diff
@@ -2843,3 +2843,22 @@ def test_obj_with_exclusions_duplicate_columns():
     result = gb._obj_with_exclusions
     expected = df.take([0, 2, 3], axis=1)
     tm.assert_frame_equal(result, expected)
+
+
+def test_groupby_numeric_only_std_no_result():
+    # GH 51080
+    dicts_non_numeric = [{"a": "foo", "b": "bar"}, {"a": "car", "b": "dar"}]
+    df = DataFrame(dicts_non_numeric)
+    dfgb = df.groupby("a")
+    result = dfgb.std(numeric_only=True)
+
+    assert result.empty
```

Comment:
Can you assert the full result here - construct `expected = DataFrame(...)` and then use `tm.assert_frame_equals(result, expected)`.

Reply:
Done.